x/ccv/provider/types/params.go

```diff
@@ -35,9 +35,9 @@ const (
 	// fraction of total voting power that the slash meter can hold.
 	DefaultSlashMeterReplenishFraction = "0.05"
 
-	// DefaultMaxPendingSlashPackets defines the default maximum amount of pending slash packets that can
-	// be queued for a consumer before the provider chain halts.
-	DefaultMaxPendingSlashPackets = 1000
+	// DefaultMaxThrottledPackets defines the default amount of throttled slash or vsc matured packets
+	// that can be queued for a single consumer before the provider chain halts.
+	DefaultMaxThrottledPackets = 1000
```

Comment:
This is an arbitrary number, right?

I'm a bit uneasy about halting the provider if 1 consumer exceeds this number.

I'm fine with having this in, provided we treat consumers with a lot less trust in the coming iterations.

Could we devise a different approach where we would remove a consumer if this ever happens? If a removal event happens the consumer chain should be able to submit a governance proposal to be reinstated in ICS.

Reply:
I'd prefer to remove the halt code altogether once we have better mechanisms to throw out spam slash packets. Removing the consumer in the event that too many slash packets are recv seems like going too far down the route of safety mechanisms, similar to https://github.com/cosmos/interchain-security/issues/547

I'm going to continue merging this PR tho, as this conversation can be continued in open issues and the changes don't directly relate to halting the provider